pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -62,8 +82,20 @@ func (c *CSIMaxVolumeLimitChecker) attachableLimitPredicate(
 		return true, nil, nil
 	}
 
+	// If either Node or CSINode objects are invalid, don't schedule pod on node
+	if nodeInfo == nil || nodeInfo.Node() == nil {
+		return false, nil, fmt.Errorf("invalid node")
+	}
+
+	// If CSINode doesn't exist, the predicate may read the limits from Node object
+	csiNode, err := c.csiNodeInfo.GetCSINodeInfo(nodeInfo.Node().Name)
+	if err != nil {
+		// TODO: in v1.18 return the error instead of just logging it
```

Comment:
This will be longer than 1.18.  CSINode only gets created automatically when migration is on by default, which will not happen until migration goes GA.

Reply:
Updated comment to: `// TODO: return the error once CSINode is created by default (2 releases)`

Reply:
@msau42 do you mean CSI migration goes to GA? Is it possible to finish the TODOs now since CSI is already GA?